### ISSUE TITLE
Uncaught Exception

### ISSUE THREAD

**Issue by Freaxed** [orcema/google-notify#1]
Hi,
I'm using google-notify with node-red-google-notify package (latest version on raspberry Pi 3+)

Time to time I'm facing this error:

`17 Nov 04:22:31 - [red] Uncaught Exception:
17 Nov 04:22:31 - TypeError: Cannot read property 'close' of null
    at PlatformSender.close (/home/pi/.node-red/node_modules/castv2-client/lib/senders/platform.js:66:15)
    at PlatformSender.<anonymous> (/home/pi/.node-red/node_modules/google-notify/google-notify.js:186:35)
    at PlatformSender.emit (events.js:198:13)
    at fn.ontimeout (/home/pi/.node-red/node_modules/castv2-client/lib/senders/platform.js:56:12)
    at Object.onceWrapper (events.js:286:20)
    at fn.emit (events.js:198:13)
    at Timeout.ontimeout [as _onTimeout] (/home/pi/.node-red/node_modules/castv2-client/lib/controllers/heartbeat.js:44:10)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
nodered.service: Main process exited, code=exited, status=1/FAILURE
nodered.service: Failed with result 'exit-code'.
nodered.service: Service RestartSec=20s expired, scheduling restart.
nodered.service: Scheduled restart job, restart counter is at 3.
`

In short, due to an heartbeat timeout,  my whole node-red is restarting.

Thanks!

**Comment by orcema** [comment]
ok, i will try to fix it this week-end

**Comment by orcema** [comment]
fix has been published on version 1.4.7

**Comment by Freaxed** [comment]
Hi,
I’m still having the same bug.. by looking at you commit I’m not sure the code is checking the right pointer.. the null object (client) is located in:
`
PlatformSender.prototype.close = function() {
  this.client.close();
}; `

I understand this is a third party library.. not sure if there is a way to check for the ‘client’ object before calling ‘close()’… another solution could be to try/catch the error..

Regards,
Andrea

**Comment by orcema** [comment]
it's difficult for me to fix this issue as it hasn't happened to me
